src/libraries/StableMath.sol

```diff
@@ -64,6 +66,8 @@ library StableMath {
         uint256 swapFee,
         uint256 N_A // solhint-disable-line var-name-mixedcase
     ) internal pure returns (uint256 dx) {
+        // overflow and underflow are not possible as reserves, amountIn <= uint104 and precision multipliers are maximum 1e18 (uint60)
+        // and swapFee < ONE_HUNDRED_PERCENT
```

Comment:
Do we check all callsites?

Reply:
just did a check. 

Seems that for `_getAmountOut` (exact in trades), it is indeed possible for the input amountIn to be more than `uint104`. But this would be caught at `_updateAndUnlock` or the overflow in `_computeLiquidityFromAdjustedBalances`. So I think there's no need to do anything. If we want to be super safe we can add a check `require(lAmountIn < type(uint104).max)` in StablePair. Currently a similar check exist for exact out trades. 

Also reviewed the analogous cases in ConstantProduct. It's not a concern for ConstantProduct even if the amountIn exceeds the range cuz the math is checked (so it reverts when overflow), whereas the math is unchecked for Stable.